examples/postgres/test_all

```diff
@@ -1,7 +1,22 @@
 #!/bin/sh
 set -e
 
-export DATABASE_URL="postgres://localhost/diesel_examples"
+set -a
+. ../../.env || true
+DATABASE_URL=${DATABASE_URL-${PG_DATABASE_URL}}
+set +a
+
+trap finish EXIT
+
+function finish {
+  # Make sure when we're done, we leave behind an empty database. Otherwise,
+  # at the time of writing, running diesel_tests stumbles as it tries to run
+  # conflicting migrations on an overlapping schema.
+  #
+  # TODO: Find a better way to avoid mixing of resources across tests.
+  ../../bin/diesel database drop
+  ../../bin/diesel database touch
```

Comment:
See above, the examples leave `diesel_test.posts` behind which clashes which the migration history use in the `diesel_tests`. Or, rather, that's what it looked like to me.

> This is not an acceptable solution.

I agree, and it's definitely going on. After all, the database is shared across all tests...

There are also two more issues I haven't even looked into yet (well, I've tried, but didn't manage):

* the `sqlite` diesel tests are sometimes built in a way that the inferred schema is empty (see the PR description), and since nothing invalidates the binary when the sqlite database changes, you need to force rebuild of the test so that it can pass
* not a schema problem, but sometimes the postgres `times_relative_to_now_encode_correctly() ` fails (i.e. the assertion fails). Again, this disappears immediately when I force a rebuild (as happens when I expand the assertion to print what's going on). This usually happens after I've had the laptop in sleep mode or when I run it the first time from a clean slate.